src/backend/vulkan/src/lib.rs

```diff
@@ -388,8 +392,15 @@ impl hal::PhysicalDevice<Backend> for PhysicalDevice {
             )
         }).unwrap();
 
+        // pipeline_cache_uuid is a uuid representing the device, so truncating it probably
+        // won't cause collisions. TODO: Return error? Very unlikely to fail.
+        debug_assert!(ash::vk::types::VK_UUID_SIZE > 8);
+        let mut reader = Cursor::new(self.properties.pipeline_cache_uuid);
+        let device_uuid = reader.read_u64::<LittleEndian>().unwrap() as usize;
```

Comment:
pipeline UUID is a physical device property iirc, therefore, would be the same for all logical devices